scripts/linux/runX509Tests.sh

```diff
@@ -0,0 +1,363 @@
+            REGISTRY_UNAME="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 12 ]; then
+            REGISTRY_PW="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 13 ]; then
+            IMAGE_TAG="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 14 ]; then
+            PROTOCOL="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 15 ]; then
+            EVENTHUB_CONNECTION_STRING="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 16 ]; then
+            EDGE_DEVICE_ID="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "-h" | "--help" ) usage;;
+                "-dir" | "--script-dir" ) save_next_arg=1;;
+                "-ic" | "--install-root-ca-cert" ) save_next_arg=2;;
+                "-ik" | "--install-root-ca-key" ) save_next_arg=3;;
+                "-pw" | "--root-ca-key-password" ) save_next_arg=4;;
+                "-qs" | "--run-quickstart" ) RUN_QUICKSTART=1;;
+                "-ld-x509ca" | "--run-leaf-device-x509-ca-auth" ) RUN_LEAF_X509_CA=1;;
+                "-ld-x509th" | "--run-leaf-device-x509-thumbprint-auth" ) RUN_LEAF_X509_THUMB=1;;
+                "-ld-sas" | "--run-leaf-device-sas-auth" ) RUN_LEAF_SAS=1;;
+                "-exe" | "--test-executable" ) save_next_arg=5;;
+                "-d" | "--device-id" ) save_next_arg=6;;
+                "-a" | "--packages" ) save_next_arg=7;;
+                "-c" | "--connection-string" ) save_next_arg=8;;
+                "-n" | "--edge-hostname" ) save_next_arg=9;;
+                "-r" | "--registry" ) save_next_arg=10;;
+                "-u" | "--registry-username" ) save_next_arg=11;;
+                "-p" | "--registry-password" ) save_next_arg=12;;
+                "-t" | "--image-tag" ) save_next_arg=13;;
+                "-proto" | "--protocol" ) save_next_arg=14;;
+                "-e" | "--eventhub-connection-string" ) save_next_arg=15;;
+                "-ed-id" | "--edge-device-id" ) save_next_arg=16;;
+                * ) usage;;
+            esac
+        fi
+    done
+
+    if [ -z ${CERT_SCRIPT_DIR} ] || [ ! -d ${CERT_SCRIPT_DIR} ]; then
+        echo "Certificate script dir is invalid"
+        print_help_and_exit
+    fi
+
+    if [ ${RUN_QUICKSTART} -eq 1 ] ||
+        [ ${RUN_LEAF_SAS} -eq 1 ] ||
+        [ ${RUN_LEAF_X509_CA} -eq 1 ] ||
+        [ ${RUN_LEAF_X509_THUMB} -eq 1 ]; then
+        if [ -z ${TEST_EXECUTABLE} ] || [ ! -f ${TEST_EXECUTABLE} ]; then
+            echo "Test executable not found, please provide a valid path to the test application"
+            print_help_and_exit
+        fi
+    fi
+}
+
+###############################################################################
+# Run the quick start test application
+###############################################################################
+function run_quickstart_as_gateway()
+{
+    local executable=${1}
+    local edge_device_id=${2}
```

Comment:
>edge_device_id [](start = 10, length = 14)

This is named "edge_device_id", but you're passing in DEVICE_ID.

Reply:
I think this can be improved. Let me clean this up for clarity